kolena/_experimental/object_detection/dataset.py

```diff
@@ -0,0 +1,246 @@
+    :param inference: Column name for inference bounding boxes
+    :param iou_threshold: The [IoU ↗](../../metrics/iou.md) threshold, defaulting to `0.5`.
+    :param threshold_strategy: The confidence threshold strategy. It can either be a fixed confidence threshold such
+        as `0.5` or `0.75`, or the F1-optimal threshold.
+    :param min_confidence_score: The minimum confidence score to consider for the evaluation. This is usually set to
+        reduce noise by excluding inferences with low confidence score.
+    """
+    labels = {x.label for x in itertools.chain.from_iterable(pred_df[ground_truth])}
+    labels.union({x.label for x in itertools.chain.from_iterable(pred_df[inference])})
+    is_multiclass = len(labels) >= 2
+    match_fn = match_inferences_multiclass if is_multiclass else match_inferences
+
+    idx = {name: i for i, name in enumerate(list(pred_df), start=1)}
+
+    all_bbox_matches: Union[List[MulticlassInferenceMatches], List[InferenceMatches]] = []
+    for record in pred_df.itertuples():
+        ground_truths = [
+            LabeledBoundingBox(box.top_left, box.bottom_right, box.label) for box in record[idx[ground_truth]]
+        ]
+        inferences = record[idx[inference]]
+        all_bbox_matches.append(
+            match_fn(  # type: ignore[arg-type]
+                ground_truths,
+                filter_inferences(inferences, min_confidence_score),
+                mode="pascal",
+                iou_threshold=iou_threshold,
+            ),
+        )
+
+    thresholds: dict[str, float]
+
+    if is_multiclass:
+        if isinstance(threshold_strategy, dict):
+            thresholds = threshold_strategy
+        elif isinstance(threshold_strategy, float):
+            thresholds = defaultdict(lambda: threshold_strategy)
+        else:
+            thresholds = compute_optimal_f1_threshold_multiclass(
+                cast(List[MulticlassInferenceMatches], all_bbox_matches),
+            )
+        results = [
+            multiclass_datapoint_metrics(cast(MulticlassInferenceMatches, matches), thresholds)
+            for matches in all_bbox_matches
+        ]
+    else:
+        if isinstance(threshold_strategy, dict) and threshold_strategy:
+            threshold = next(iter(threshold_strategy.values()))
+        elif isinstance(threshold_strategy, float):
+            threshold = threshold_strategy
+        else:
+            threshold = compute_optimal_f1_threshold(cast(List[InferenceMatches], all_bbox_matches))
+        results = [
+            single_class_datapoint_metrics(cast(InferenceMatches, matches), threshold) for matches in all_bbox_matches
+        ]
+
+    return pd.concat([pd.DataFrame(results), pred_df], axis=1)
+
+
+def _validate_column_present(df: pd.DataFrame, col: str) -> None:
+    if col not in df.columns:
+        raise IncorrectUsageError(f"Missing column '{col}'")
+
+
+def upload_object_detection_results(
+    dataset_name: str,
```

Comment:
When we use this function to write tests, we are going to face an issue because we use **dataset_name** both to download the dataset and to upload the results. Most likely, the **dataset_name** used for uploading the results will have a prefix.

So, what do you think about adding a new dataset_name parameter to distinguish between the usages of dataset_name between: 
`dataset_df = dataset.download_dataset(dataset_name)
`
and 
`dataset.upload_results(dataset_name, model_name, [(eval_config, results_df)])
`